source/codegen/templates/service_helpers.mako

```diff
@@ -19,7 +19,10 @@ ${initialize_input_params(function_name, parameters)}
       };
       uint32_t session_id = 0;
       const std::string& session_name = request->session_name();
-      auto cleanup_lambda = [&] (uint32_t id) { library_->${config['close_function']}(id); };
+<%
```

Comment:
[Optional] Consider moving the contents of this `<%` block to the one for the entire def around line 13. It might help the readability to not break up the template code with the python code when possible.

Reply:
This is made consistent with DMM, so maybe in future PRs we can move it.

Reply:
If this helps with readability, I think we can move it now. When we merge feature branches with `main` we will have to be careful to select best implementation.

Reply:
Got it. Doing it right now.